Add centerPointLabels option for linear radial scale

t-mangoe opened the pull request:
I added code to automatically center the point labels when using a polar graph.
This solves the issue of #9662

LeeLenaleee commented on src/scales/scale.radialLinear.js:
```diff
@@ -404,8 +405,8 @@ export default class RadialLinearScale extends LinearScaleBase {
 
-  getPointPosition(index, distanceFromCenter) {
-    const angle = this.getIndexAngle(index) - HALF_PI;
+  getPointPosition(index, distanceFromCenter, {additionalAngle = 0} = {}) {
```
Since you only expect a singel value as extra argument it would be better for performance to just pass that value instead of a object that has to be destructured.

t-mangoe replied:
I changed the argument to a single value.

LeeLenaleee reviewed (commented):
The documentation for the new properties and typings are also still missing

kurkle reviewed (changes requested):
All the points @LeeLenaleee pointed need to be changed before we can approve this.

t-mangoe commented on src/controllers/controller.polarArea.js:
```diff
@@ -205,7 +205,8 @@ PolarAreaController.overrides = {
       pointLabels: {
-        display: false
+        display: false,
+        polarChartLabel: true
```
I changed the option name to `centerPointLabels`.
Also, I changed the default value to false.

t-mangoe commented on src/scales/scale.radialLinear.js:
```diff
@@ -265,7 +266,7 @@ function numberOrZero(param) {
 
 function createPointLabelContext(parent, index, label) {
-  return createContext(parent, {
+  return Object.assign(Object.create(parent), {
```
Sorry. I did't mean to change it.
I reverted it back.

kurkle reviewed (commented):
The implementation and test looks good to me.
There are couple of things to do still:

- add to documentation [1]
- add to types [2]
- add a sample (optional, but would be good)

1)
https://github.com/chartjs/Chart.js/blob/5bea389c42b949ca6ebdabf1706295d126ea5f4b/docs/axes/radial/linear.md#L119-L128

2)
https://github.com/chartjs/Chart.js/blob/5bea389c42b949ca6ebdabf1706295d126ea5f4b/types/index.esm.d.ts#L3315

t-mangoe commented:
@kurkle 
Thank you for your kind instruction!!
I added documentation and types.
(I didn't know how to add a sample, so I added no samples. sorry...)

LeeLenaleee commented:
Easyest way to add a sample for this is to copy paste [this file](https://github.com/chartjs/Chart.js/blob/master/docs/samples/other-charts/polar-area.md) and name it something like: `polar-area-centerPointLabels`

After that you enable the point labels in the options (config block).

Then you can add that sample in the navbar by adding it beneath the link for the default polar area sample [here](https://github.com/chartjs/Chart.js/blob/0d250a1fbf117a826d9318bb308dd620c1357358/docs/.vuepress/config.js#L166).

That should make it work. You can then test it by running `npm run docs:dev` and after it is done building by browsing to `localhost:8080/samples` and then in the sidebar to the sample you just made.

If it doesn't work I will make the sample after this pr gets merged so no big deal

kurkle commented:
@LeeLenaleee if you have the time to create a sample for this, it would be great!

kurkle commented:
Thanks @t-mangoe
